src/RedisClient.php

```diff
@@ -62,6 +62,11 @@ private function connectIfNeeded(): void
                 $this->redis->getLastError() ?? 'unknown error'
             ));
         }
+        if($params->getUsername() && $params->getPassword()) {
+            $this->redis->auth([$params->getUsername(), $params->getPassword()]);
+        } else if ($params->getPassword()) {
+            $this->redis->auth([$params->getPassword()]);
+        }
```

Comment:
Hi @AmitKannaujiya , I think the build is failing because psalm thinks that the `auth()` method can be only called passing a string, while it's valid to pass both a string and an array. What I would suggest is to suppress the issue in the first case, and pass a string in the second case:


```suggestion
        if($params->getUsername() && $params->getPassword()) {
            /** @psalm-suppress InvalidArgument */
            $this->redis->auth([$params->getUsername(), $params->getPassword()]);
        } else if ($params->getPassword()) {
            $this->redis->auth($params->getPassword());
        }
```
What do you think?